Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Extensions/ApiGatewayResponseExtensionsTests.cs

```diff
@@ -0,0 +1,110 @@
+            // Arrange
+            HttpResponse httpResponse = ((APIGatewayProxyResponse)testCase.Response).ToHttpResponse(ApiGatewayEmulatorMode.HttpV1);
+
+            // Assert
+            testCase.Assertions(httpResponse, ApiGatewayEmulatorMode.HttpV1);
+        }
+
+        [Theory]
+        [MemberData(nameof(ApiGatewayTestCases.V1TestCases), MemberType = typeof(ApiGatewayTestCases))]
+        public void ToHttpResponse_ConvertsCorrectlyV1Rest(string testName, ApiGatewayTestCase testCase)
+        {
+            // Arrange
+            HttpResponse httpResponse = ((APIGatewayProxyResponse)testCase.Response).ToHttpResponse(ApiGatewayEmulatorMode.Rest);
+
+            // Assert
+            testCase.Assertions(httpResponse, ApiGatewayEmulatorMode.Rest);
+        }
+
+        [Theory]
+        [MemberData(nameof(ApiGatewayTestCases.V2TestCases), MemberType = typeof(ApiGatewayTestCases))]
+        public void ToHttpResponse_ConvertsCorrectlyV2(string testName, ApiGatewayTestCase testCase)
+        {
+            // Arrange
+            HttpResponse httpResponse = ((APIGatewayHttpApiV2ProxyResponse)testCase.Response).ToHttpResponse();
+
+            // Assert
+            testCase.Assertions(httpResponse, ApiGatewayEmulatorMode.HttpV2);
+        }
+
+        [Fact]
+        public void ToHttpResponse_APIGatewayHttpApiV2ProxyResponse_InfersResponseFormatForValidJson()
+        {
+            var jsonBody = "{\"key\":\"value\"}";
+            var apiResponse = new APIGatewayHttpApiV2ProxyResponse
+            {
+                Body = jsonBody,
+                StatusCode = 0 // No status code set
+            };
+
+            var httpResponse = apiResponse.ToHttpResponse();
+
+            Assert.Equal(200, httpResponse.StatusCode);
+            Assert.Equal("application/json", httpResponse.ContentType);
+
+            httpResponse.Body.Seek(0, SeekOrigin.Begin);
+            using var reader = new StreamReader(httpResponse.Body);
+            var bodyContent = reader.ReadToEnd();
+            Assert.Equal(jsonBody, bodyContent);
+        }
+
+        [Fact]
+        public void ToHttpResponse_APIGatewayHttpApiV2ProxyResponse_InfersResponseFormatForValidJson2()
+        {
+            var jsonBody = "hello lambda";
+            var apiResponse = new APIGatewayHttpApiV2ProxyResponse
+            {
+                Body = jsonBody,
+                StatusCode = 0 // No status code set
+            };
+
+            var httpResponse = apiResponse.ToHttpResponse();
+
+            Assert.Equal(200, httpResponse.StatusCode);
+            Assert.Equal("application/json", httpResponse.ContentType);
+
+            httpResponse.Body.Seek(0, SeekOrigin.Begin);
+            using var reader = new StreamReader(httpResponse.Body);
+            var bodyContent = reader.ReadToEnd();
+            Assert.Equal(jsonBody, bodyContent);
+        }
+
+        [Fact]
+        public void ToHttpResponse_APIGatewayHttpApiV2ProxyResponse_HandlesNonJsonResponse()
```

Comment:
im not actually convinced this test case is right since i have not been able to return a non json response from a lambda (lol). Even if i return this `{fdasfdsa}`, it thinks the string is valid json. 

The docs are not really clear either
https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html

With the 2.0 format version, API Gateway can infer the response format for you. API Gateway makes the following assumptions if your Lambda function returns valid JSON and doesn't return a statusCode:

isBase64Encoded is false.

statusCode is 200.

content-type is application/json.

body is the function's response.


from the examples they have, they even consider a regular string to be valid. so im thinking i need to update the logic with the json validator/remove it (see other comment)

Reply:
To be clear you are saying you can't return a non-json response when the status code is not set correct? I have definitely seen non-JSON response with Http Api V2 but that I'm sure has been with the status code set.

Reply:
i mean i cant figure out how to return a non json response from a lamda function. if i do something like

```

return {}
```

this is valid json. but even doing 

```
return "abc"
```

from the lambda, api gateway thinks this is valid json.

and i have also tried (in javascript) doing

```
callback(1234)
```

do send the direct value, but the content type is still set as application/json